broker/lead.go

```diff
@@ -115,6 +109,41 @@ func lead(authToken models.AuthToken, policy *models.Policy) error {
 	return err
 }
 
+func checkIfPolicyIsLead(policy *models.Policy) error {
+	var recoveredPolicy models.Policy
+	log.Printf("[lead] found lead for existing policy %s", policy.Uid)
+
+	policyDoc, err := lib.GetFirestoreErr(models.PolicyCollection, policy.Uid)
+	if err != nil {
+		log.Printf("[lead] error getting policy %s from firebase: %s", policy.Uid, err.Error())
+		return nil
+	} else if !policyDoc.Exists() {
+		log.Printf("[lead] policy %s not found on Firebase", policy.Uid)
+		return nil
+	}
+
+	if err := policyDoc.DataTo(&recoveredPolicy); err != nil {
+		log.Printf("[lead] error converting policy %s data: %s", policy.Uid, err.Error())
+		return nil
+	}
+
+	if recoveredPolicy.Status != models.PolicyStatusPartnershipLead && recoveredPolicy.Status != models.PolicyStatusInitLead {
+		log.Printf("[lead] error policy %s is not a lead", policy.Uid)
+		return errors.New("policy is not a lead")
+	}
+
+	policy.CreationDate = recoveredPolicy.CreationDate
+	policy.Updated = recoveredPolicy.Updated
```

Comment:
updated field can be removed as it will be set to now if the changes are successful

Reply:
Done in https://github.com/wopta/goworkspace/pull/121/commits/0d1cfbfbdb3a1bfc445c793fda5ffeef4f83cc4c